lib/buffer.js

```diff
@@ -246,18 +247,30 @@ function fromArrayLike(obj) {
   const length = obj.length;
   const b = allocate(length);
   for (var i = 0; i < length; i++)
-    b[i] = obj[i] & 255;
+    b[i] = obj[i];
   return b;
 }
 
 function fromArrayBuffer(obj, byteOffset, length) {
+  if (!isArrayBuffer(obj))
+    throw new TypeError('argument is not an ArrayBuffer');
```

Comment:
> I assume the `isArrayBuffer` check was only added in the first place to avoid crashes on the C++ side, but that doesn’t seem to be a concern anymore if everything’s done in JS.

Question is if we want to enforce an actual `ArrayBuffer`, or if users could passed a munged up object that looks like one. Personally I prefer to leave these in where possible b/c it makes code transitions between C++ and JS smoother (buffer has bounced all over the place in the last 3 years).